File: src/main/java/net/atos/zac/app/identity/model/RESTGroup.java
/*
 * SPDX-FileCopyrightText: 2021 Atos
 * SPDX-License-Identifier: EUPL-1.2+
 */

package net.atos.zac.app.identity.model;

import jakarta.validation.constraints.Max;
import jakarta.validation.constraints.NotNull;

public class RESTGroup {
    /**
     * Since this is used for the 'identificatie' field in
     * {@link net.atos.client.zgw.zrc.model.OrganisatorischeEenheid}
     * we need to make sure it adheres to the same constraints.
     */
    @NotNull
    @Max(24)
    public String id;

    /**
     * Since this is used for the 'naam' field in
     * {@link net.atos.client.zgw.zrc.model.OrganisatorischeEenheid}
     * we need to make sure it adheres to the same constraints.
     */
    @NotNull
    @Max(50)
    public String naam;
}

File: src/main/kotlin/net/atos/zac/app/identity/model/RestGroup.kt
/*
 * SPDX-FileCopyrightText: 2024 Lifely
 * SPDX-License-Identifier: EUPL-1.2+
 */
package net.atos.zac.app.identity.model

import jakarta.validation.constraints.Size
import nl.info.zac.identity.model.Group
import nl.info.zac.util.AllOpen
import nl.info.zac.util.NoArgConstructor

@NoArgConstructor
@AllOpen
data class RestGroup(
    /**
     * Since this is used for the 'identificatie' field in
     * [net.atos.client.zgw.zrc.model.OrganisatorischeEenheid]
     * we need to make sure it adheres to the same constraints.
     */
    @field:Size(max = 24)
    var id: String,

    /**
     * Since this is used for the 'naam' field in
     * [net.atos.client.zgw.zrc.model.OrganisatorischeEenheid]
     * we need to make sure it adheres to the same constraints.
     */
    @field:Size(max = 50)
    var naam: String
)

fun Group.toRestGroup(): RestGroup =
    RestGroup(this.id, this.name)

fun List<Group>.toRestGroups(): List<RestGroup> =
    this.map { it.toRestGroup() }
